library/core/src/ptr/const_ptr.rs

```diff
@@ -7,6 +7,7 @@ use safety::{ensures, requires};
 
 #[cfg(kani)]
 use crate::kani;
+use core::mem;
 
 impl<T: ?Sized> *const T {
```

Comment:
Is the title of this PR — "Contract and Harnesses for `<*mut T>::byte_add`, `byte_sub` and `byte_offset`" — correct? It looks like your changes all occur in `const_ptr.rs`, not `mut_ptr.rs`.

Reply:
Partially yes, it was opened as a PR for `<*mut T>::byte_add`, `byte_sub` and `byte_offset`. But later, Carolyn asked me to merge const methods too as their contracts were quite similar. Now, the PR has both const and mut versions of the functions.